src/main/scala/net/snowflake/spark/snowflake/io/CloudStorageOperations.scala

```diff
@@ -1839,14 +1859,15 @@ case class InternalGcsStorage(param: MergedParameters,
         ///////////////////////////////////////////////////////////////////////
     }
 
+    val result = fileUploadResults.collect().toList
+
     val endTime = System.currentTimeMillis()
     CloudStorageOperations.log.info(
       s"""${SnowflakeResultSetRDD.MASTER_LOG_PREFIX}:
          | Finish uploading data for ${data.getNumPartitions} partitions in
          | ${Utils.getTimeString(endTime - startTime)}.
          |""".stripMargin.filter(_ >= ' '))
-
-    fileUploadResults.collect().toList
+    result
```

Comment:
The change for line 1862 - 1870 is optional for GCP support, but it is necessary to make logging time to be accurate.

The spark executors are done after `fileUploadResults.collect()` is returned. So, I change it before logging.